group/views.py

```diff
@@ -20,140 +20,166 @@
-
-    annowence_list = [
-        {'id':'1', 'name':'annowence_1', 'op':'drowsy', 'time':'20111010', 'content':'asdasd'},
-        {'id':'2', 'name':'annowence_2', 'op':'drowsy', 'time':'20140505', 'content':'qwefad'},
-    ]
-
-    student_list =[
-        {'name': 'Tom', 'time': '2014/1/2 23:00'},
-        {'name': 'Amy', 'time': '2014/1/2 23:00'},
-        {'name': 'Jess', 'time': '2014/1/2 23:00'},
-        {'name': 'Bieber', 'time': '2014/1/2 23:00'},
-        {'name': 'Ian', 'time': '2014/1/2 23:00'},
-        {'name': 'Anthony', 'time': '2014/1/2 23:00'},
-        {'name': 'Lily', 'time': '2014/1/2 23:00'}
-    ]
-
-    ta_list =[
-        {'name': 'drowsy'},
-        {'name': 'henry'},
-    ]
+def detail(request, group_id):
+    
+    try:
+        group = Group.objects.get(id = group_id)
+    except:
+        logger.warning('Group: group does not exist - gid: %s!' % group_id)
+        raise Http404('Group does not exist')
+
+    all_contest = group.trace_contest.all()[0:5]
+    annowence_list = group.announce.all()[0:5]
+    student_list = group.member.all()
+    coowner_list = group.coowner.all()
+    owner = group.owner
+
+    running_contest_list = []
+    ended_contest_list = []
+    now = timezone.now()
+    for contest in all_contest:
+        if contest.start_time < now and contest.end_time > now:
+            running_contest_list.append(contest)
+        elif contest.end_time < now:
+            ended_contest_list.append(contest)
 
     return render(
         request, 'group/groupDetail.html', {
             'rc_list': running_contest_list, 
             'ec_list': ended_contest_list,
             'an_list': annowence_list,
-            'ta_list': ta_list,
+            'coowner_list': coowner_list,
+            'owner': owner,
             's_list': student_list,
-            'group_name': 'GGgroup', 
-            'group_description': 'blablabla, this is a description',
+            'group_name': group.gname, 
+            'group_description': group.description,
+            'group_id': group.id,
         })
 
 
 def list(request):
-    group_list = [
-        {'name': 'GGgroup', 'ta_name': 'drowsy, henry', 'mem_number': '42', 'gid': '001'},
-        {'name': '2ndgroup', 'ta_name': 'TA1', 'mem_number': '1', 'gid': '002'},
-        {'name': '3rdgroup', 'ta_name': 'TA2, henry', 'mem_number': '22', 'gid': '003'},
-        {'name': '4thgroup', 'ta_name': 'drowsy, TA3', 'mem_number': '34', 'gid': '004'},
-        {'name': '5thgroup', 'ta_name': 'TA4', 'mem_number': '105', 'gid': '005'},
-        {'name': '6thgroup', 'ta_name': 'T5A', 'mem_number': '42', 'gid': '006'},
-    ]
+
+    group_list = Group.objects.order_by('-creation_time')
     return render(
         request,'group/groupList.html', {
             'g_list': group_list
         })
+
+def new(request):
+    if request.method == 'GET':
+            form = GroupForm()
+            return render(request,'group/editGroup.html',{'form':form})
+    if request.method == 'POST':
```

Comment:
Sorry for ambiguous, in that case, I mean we don't need a log for failure creation. We need it for successful one.